hw/top_earlgrey/util/top_earlgrey_reduce.py

```diff
@@ -77,7 +134,8 @@ def main():
 
     # Regenerate boot ROM
     print("Regenerating boot ROM...")
-    cmd = ["ninja", "-C", top_path + "/build-out", "sw/device/boot_rom/boot_rom_export_fpga_nexysvideo"]
+    cmd = ["ninja", "-C", top_path + "/build-out",
+           "sw/device/boot_rom/boot_rom_export_fpga_nexysvideo"]
```

Comment:
That's the command failing in CI:
```
INFO: Updating target to nexysvideo settings
INFO: Flash to 128 pages
INFO: Memory to 64KB
ERROR: Failed to regenerate boot ROM: Command '['ninja', '-C', './build-out', 'sw/device/boot_rom/boot_rom_export_fpga_nexysvideo']' returned non-zero exit status 1.
ERROR: ninja: Entering directory `./build-out'
ninja: fatal: chdir to './build-out' - No such file or directory
```
I would suggest to do the following:
1. Add an argument to trigger the boot ROM generation or skip it. In CI, we would skip it.
2. Add an additional step to the `sw_build` job in CI (after the current steps) that calls this script and with boot ROM generation. Then, once `sw_build` is done, we have the boot ROM for the Nexysvideo FPGA. For this, you also need to change `/build_out` to `"$OBJ_DIR"` (points to build-out in CI). Maybe for CI we also need the this value also needs to be an argument.
3. Then during bitstream generation, we just launch the script without boot ROM generation in CI.

Maybe @imphil has a better idea.

Reply:
Don't build software in this script. Run the script before the SW build and before the hardware build. If we additionally want a software build without the size reduction (which we probably want, since we want DV and Verilator regressions to use the unmodified flash sizes), we need to duplicate the SW build job.

Reply:
thanks all, i completely forgot that nexysvideo is a different target type for all builds. This should make things much easier.